tests/src/Kernel/Entity/FieldAccessTest.php

```diff
@@ -0,0 +1,189 @@
+use Drupal\og\Og;
+use Drupal\Component\Utility\Unicode;
+use Drupal\node\Entity\NodeType;
+use Drupal\user\Entity\User;
+use Drupal\user\Entity\Role;
+use Drupal\field\Entity\FieldStorageConfig;
+
+/**
+ * Tests OG specific field access.
+ *
+ * @group og
+ */
+class FieldAccessTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['system', 'user', 'field', 'entity_reference', 'node', 'og'];
+
+  /**
+   * @var string
+   *
+   * The machine name of the group node type.
+   */
+  protected $groupBundle;
+
+  /**
+   * @var \Drupal\user\Entity\User
+   */
+  protected $adminUser;
+
+  /**
+   * @var \Drupal\user\Entity\User
+   */
+  protected $authenticatedUser;
+
+  /**
+   * @var \Drupal\Core\Entity\EntityAccessControlHandlerInterface
+   */
+  protected $userAccessControlHandler;
+
+  /**
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $fieldDefinition;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Add membership and config schema.
+    $this->installConfig(['og']);
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installSchema('system', 'sequences');
+
+    // Setting up variables.
+    $this->groupBundle = Unicode::strtolower($this->randomMachineName());
+
+    // Create a group.
+    NodeType::create([
+      'type' => $this->groupBundle,
+      'name' => $this->randomString(),
+    ])->save();
+
+    // Define the group content as group.
+    Og::groupManager()->addGroup('node', $this->groupBundle);
+
+    // Add og audience field to users.
+    Og::createField(OG_AUDIENCE_FIELD, 'user', 'user');
+
+    Role::create(['id' => 'group_admin', 'label' => 'Group Admin'])
+      ->grantPermission('administer group')
+      ->grantPermission('administer users')
+      ->save();
+
+    // Create two users.
+    $this->adminUser = User::create(['name' => $this->randomString()]);
+    $this->adminUser->addRole('group_admin');
+    $this->adminUser->save();
+
+    $this->authenticatedUser = User::create(['name' => $this->randomString()]);
+    $this->authenticatedUser->save();
+
+    $this->fieldDefinition = $this->adminUser->getFieldDefinition(OG_AUDIENCE_FIELD);
+    $this->userAccessControlHandler = \Drupal::entityManager()->getAccessControlHandler('user');
+  }
+
+  /**
+   * Test anonymous users.
+   *
+   * @dataProvider providerTestAnonymousUserAccess
+   *
+   * @param string $operation
+   */
+  public function testAnonymousUserAccess($operation) {
+    $this->assertTrue($this->userAccessControlHandler->fieldAccess($operation, $this->fieldDefinition, new AnonymousUserSession()));
+  }
+
+  /**
+   * Data provider for testAnonymousUserAccess.
+   *
+   * @return array
+   */
+  public function providerTestAnonymousUserAccess() {
+    return [
+      ['edit'],
+      ['view'],
+      ['delete'],
+    ];
+  }
```

Comment:
I was very confused for a moment seeing that anonymous users have access to edit, view and delete any OG audience field, but this appears to allow the field to be added to a new user account on registration. In fact our `hook_entity_field_access()` implementation just returns neutral in this case, and the field module itself allows it.